logstreams/src/main/java/io/zeebe/logstreams/storage/atomix/AtomixLogStorageReader.java

```diff
@@ -75,31 +66,17 @@ public long read(final DirectBuffer readBuffer, final long address) {
     return entry.index() + 1;
   }
 
-  /**
-   * This is currently a quite slow implementation as Atomix does not support navigating backwards;
-   * it would require a refactor there if this is ever too slow.
-   *
-   * <p>{@inheritDoc}
-   */
   @Override
   public long readLastBlock(final DirectBuffer readBuffer) {
-    final var firstIndex = reader.getFirstIndex();
-    var index = reader.getLastIndex();
-
-    do {
-      reader.reset(index);
-      if (!reader.hasNext()) {
-        break;
-      }
+    reader.seekToAsqn(Long.MAX_VALUE);
```

Comment:
Optional: Catch the exception incase we seek to non-committed. And use the previous logic of reading backwards as backup.